trace-sdk/src/test/java/io/bitrise/trace/data/application/ApplicationVersionCodeDataCollectorTest.java

```diff
@@ -0,0 +1,52 @@
+package io.bitrise.trace.data.application;
+
+import android.content.Context;
+
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import io.bitrise.trace.BuildConfig;
+import io.bitrise.trace.configuration.ConfigurationManager;
+import io.bitrise.trace.data.collector.application.ApplicationVersionCodeDataCollector;
+import io.bitrise.trace.data.dto.Data;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for {@link ApplicationVersionCodeDataCollector}
+ */
+public class ApplicationVersionCodeDataCollectorTest {
+
+    private final static int versionCode = 123;
+    private final Context mockContext = Mockito.mock(Context.class);
+    private final ApplicationVersionCodeDataCollector collector = new ApplicationVersionCodeDataCollector(
+            mockContext);
+
+    @Test
+    public void collectData_configurationManagerInitialised() {
```

Comment:
Similarly to [this](https://github.com/bitrise-io/trace-android-sdk/pull/199/files#r635145991) comment, this verifies that we get the data from the ConfigurationManager. What happens if configurationManager returns null or a non valid value during operation? This test will still pass, but we will have issues. I think we should check we get a valid version code instead.

Reply:
As this is a unit test, we don't actually have a version_code or version_name available which is why i'm creating a debug configurationManager with those set.

 I had added a comment (which I later deleted) that said something about if the configurationManager doesn't have the version_code or version_name available, it will crash the customers application. I had instrumented tests for this - but i decided they weren't actually very useful to assume a crash. But I didn't think refactoring the behaviour here made sense either.

My suggestion would be - we should re-prioritise APM-2424 (higher priority?) - we should update that ticket to also add tests for the ConfigurationManager around finding things that aren't available and have an agreed expected behaviour (empty string or null, not a crash).  If that makes sense?

Reply:
Sure, we can add this to that ticket. But I believe these tests just add some comfort by increasing the coverage numbers, but have no real value.

Reply:
So i think these tests have value - they're comfort checking that given an input we get an expected output as there's some processing in the middle (it would be easy to set a different content for example, maybe get the NAME/CODE properties mixed up). Unit tests are great for us so we can be confident that nothing in that process changes in the future.

Reply:
I believe it would make more sense if we would check it with real data, because this will never fail, if the config item name changes from "VERSION_CODE" to something else in the ConfigurationManager, but not in the Collector. But agree, it meets the acceptance criteria...

Reply:
So i think there's an interesting scenario if we get integrated into someone else's library - libraries don't get version_codes anymore - so this would currently cause a crash. IMO we need to deal with this scenario better - defaulting to an empty string / or unknown instead of a crash :)